### PR TITLE
Add example for cross-repo links

### PR DESCRIPTION
**Pull request by knylander-grafana**
Updated the cross-repository link example and the references page.

### CODE REVIEW

File: docs/sources/writing-guide/references/index.md [hunk 1]
```diff
@@ -15,11 +15,7 @@ keywords:
...
-> Hugo references addressed across different products' docs, such as from `/docs/grafana/` to `/docs/loki/` and vice-versa, as well as references from docs addressed to other Hugo-published content on grafana.com, can also be unpredictably addressed.
->
-> To avoid broken links in these situations on grafana.com, use regular Markdown link syntax (`[link text](/docs/repo/version/folder/file/)`) instead of Hugo references. To ensure the links work in local builds, staging environments, and the live website, you **shouldn't** use a fully qualified URL with `https://grafana.com` for links to other content on grafana.com.
-> Unlike references, Hugo does _not_ confirm that these link destinations exist during its build, so manually confirm that the published links in a local build and on the published website point correctly.
+>**Note**: When linking to specific versions or across repositories, use standard markdown links. Read the [Versions and cross-pository linking]({{< relref "#versions-and-cross-repository-linking" .}}) section for details.
```
Comment (knylander-grafana) [inline docs/sources/writing-guide/references/index.md:18]:
```suggestion
>**Note**: When linking to specific versions or across repositories, use standard markdown links. Read the [Versions and cross-pository linking]({{< relref "#versions-and-cross-repository-linking" >}}) section for details.
```

File: docs/sources/writing-guide/references/index.md [hunk 2]
```diff
@@ -81,6 +77,25 @@ You can refer to the table below for all file path based relrefs between these f
 | `/docs/sources/branch/other.md`  | `/docs/sources/branch/_index.md` | `{{</* relref "_index.md" */>}}`           |
 | `/docs/sources/branch/other.md`  | `/docs/sources/leaf/index.md`    | `{{</* relref "../leaf/index.md" */>}}`    |
 
+
+## Versions and cross-repository linking
+
+For Hugo's purposes, you can't address other versions of the docs, such as a version-specific archived docs set (`https://grafana.com/docs/grafana/v8.5/`, etc.) or `/next/` docs for links in content residing in `/latest/`, using Hugo references.
```
Comment (GrafanaWriter) [inline docs/sources/writing-guide/references/index.md:83]:
```suggestion
For Hugo's purposes, you can't address other versions of the docs, such as a version-specific archived docs set (`https://grafana.com/docs/grafana/v8.5/` and so forth) or `/next/` docs for links in content residing in `/latest/`, using Hugo references.
```

Comment (GrafanaWriter) [reply docs/sources/writing-guide/references/index.md:83]: nit :)

**Review by GrafanaWriter — APPROVED** [review]
Thank you

File: docs/sources/writing-guide/references/index.md [hunk 3]
```diff
@@ -15,11 +15,7 @@ keywords:
...
-> Hugo references addressed across different products' docs, such as from `/docs/grafana/` to `/docs/loki/` and vice-versa, as well as references from docs addressed to other Hugo-published content on grafana.com, can also be unpredictably addressed.
->
-> To avoid broken links in these situations on grafana.com, use regular Markdown link syntax (`[link text](/docs/repo/version/folder/file/)`) instead of Hugo references. To ensure the links work in local builds, staging environments, and the live website, you **shouldn't** use a fully qualified URL with `https://grafana.com` for links to other content on grafana.com.
-> Unlike references, Hugo does _not_ confirm that these link destinations exist during its build, so manually confirm that the published links in a local build and on the published website point correctly.
+>**Note**: When linking to specific versions or across repositories, use standard markdown links. Read the [Versions and cross-pository linking]({{< relref "#versions-and-cross-repository-linking" >}}) section for details.
```
Comment (imatwawana) [inline docs/sources/writing-guide/references/index.md:18]: Not sure if you mean for this to say "cross-repository" or not
```suggestion
>**Note**: When linking to specific versions or across repositories, use standard markdown links. Read the [Versions and cross-repository linking]({{< relref "#versions-and-cross-repository-linking" >}}) section for details.
```

File: docs/sources/writing-guide/references/index.md [hunk 4]
```diff
@@ -81,6 +77,25 @@ You can refer to the table below for all file path based relrefs between these f
...
+
+## Versions and cross-repository linking
+
+For Hugo's purposes, you can't address other versions of the docs, such as a version-specific archived docs set (`https://grafana.com/docs/grafana/v8.5/`, etc.) or `/next/` docs for links in content residing in `/latest/`, using Hugo references.
+
+Hugo references addressed across different products' docs, such as from `/docs/grafana/` to `/docs/loki/` and vice-versa, as well as references from docs addressed to other Hugo-published content on grafana.com, can also be unpredictably addressed.
+
+To avoid broken links in these situations on grafana.com, use regular Markdown link syntax (`[link text](/docs/repo/version/folder/file/)`) instead of Hugo references. To ensure the links work in local builds, staging environments, and the live website, you **shouldn't** use a fully qualified URL with `https://grafana.com` for links to other content on grafana.com.
+
+For cross-repository links, use a standard markdown link, like this:
```
Comment (imatwawana) [inline docs/sources/writing-guide/references/index.md:89]: Wondering if before we lay out the example if we can add the structure. Something like: 

```suggestion
For cross-repository links, use a standard markdown link, with the link structured like so:
`\docs\repo\page\`

For example:
```

**Review by imatwawana — APPROVED** [review]
Thank you for restructuring/updating this! I added a couple questions/suggestions.

**Comment by jdbaldry** [comment]
I think this should be revisited after https://github.com/grafana/writers-toolkit/pull/225 has been merged.

**Comment by knylander-grafana** [comment]
@jdbaldry Do you want me to hold off merging for now?

**Comment by jdbaldry** [comment]
I think we should be trying to use relrefs for cross repository links and we should be working on making CI support that link checking.

I'll add more specific comments on the patch.

File: docs/sources/writing-guide/references/index.md [hunk 5]
```diff
@@ -81,6 +77,28 @@ You can refer to the table below for all file path based relrefs between these f
 | `/docs/sources/branch/other.md`  | `/docs/sources/branch/_index.md` | `{{</* relref "_index.md" */>}}`           |
 | `/docs/sources/branch/other.md`  | `/docs/sources/leaf/index.md`    | `{{</* relref "../leaf/index.md" */>}}`    |
 
+
+## Versions and cross-repository linking
+
+For Hugo's purposes, you can't address other versions of the docs, such as a version-specific archived docs set (`https://grafana.com/docs/grafana/v8.5/` and so forth) or `/next/` docs for links in content residing in `/latest/`, using Hugo references.
```
Comment (jdbaldry) [inline docs/sources/writing-guide/references/index.md:83]: This isn't a Hugo thing, this is a limitation of our local webserver environments.

With the `make-docs` script, it is now possible to build multiple documentation sets together and I have demonstrated cross project link checking in https://github.com/grafana/tempo/pull/2299.

File: docs/sources/writing-guide/references/index.md [hunk 6]
```diff
@@ -81,6 +77,28 @@ You can refer to the table below for all file path based relrefs between these f
...
+
+Hugo references addressed across different products' docs, such as from `/docs/grafana/` to `/docs/loki/` and vice-versa, as well as references from docs addressed to other Hugo-published content on grafana.com, can also be unpredictably addressed.
```
Comment (jdbaldry) [inline docs/sources/writing-guide/references/index.md:85]: What does this mean?
Hugo relrefs require a specific way of specifying the parameter but they shouldn't behave unpredictably.

File: docs/sources/writing-guide/references/index.md [hunk 7]
```diff
@@ -81,6 +77,28 @@ You can refer to the table below for all file path based relrefs between these f
...
+For cross-repository links, use a standard markdown link, with the link structured like so:
+`\docs\repo\page\`
+
+For example:
+
+```markdown
+This is an [example cross-repository link](/docs/grafana/whatsnew) to the Grafana repository.
+```
+
+Using a Hugo `relref` in a cross-repository link or a link to a specific version can result in a page not found error message when running `make docs`.
```
Comment (jdbaldry) [inline docs/sources/writing-guide/references/index.md:98]: Only if the linked content isn't also mounted when using the `make-docs` script.

File: docs/sources/writing-guide/references/index.md [hunk 8]
```diff
@@ -81,6 +77,28 @@ You can refer to the table below for all file path based relrefs between these f
...
+
+Unlike references, Hugo does _not_ confirm that these link destinations exist during its build, so manually confirm that the published links in a local build and on the published website point correctly.
```
Comment (jdbaldry) [inline docs/sources/writing-guide/references/index.md:100]: With partial URIs, you also cannot check these links without the content mounted. `/docs/grafana/latest/` from `/docs/tempo/latest` won't resolve unless you have both projects mounted in the webserver.

File: docs/sources/writing-guide/references/index.md (hunk 5 above)
Comment (knylander-grafana) [reply docs/sources/writing-guide/references/index.md:83]:
```suggestion
For Grafana's webserver environments, you can't address other versions of the docs, such as a version-specific archived docs set (`https://grafana.com/docs/grafana/v8.5/` and so forth) or `/next/` docs for links in content residing in `/latest/`, using Hugo references.
```

File: docs/sources/writing-guide/references/index.md (hunk 6 above)
Comment (knylander-grafana) [reply docs/sources/writing-guide/references/index.md:85]:
```suggestion
Hugo references addressed across different products' docs, such as from `/docs/grafana/` to `/docs/loki/` and vice-versa, as well as references from docs addressed to other Hugo-published content on grafana.com, can also be predictably addressed.
```

File: docs/sources/writing-guide/references/index.md (hunk 8 above)
Comment (knylander-grafana) [reply docs/sources/writing-guide/references/index.md:100]: I will add what you noted :) 
```suggestion
Unlike references, Hugo does _not_ confirm that these link destinations exist during its build, so manually confirm that the published links in a local build and on the published website point correctly.
With partial URIs, you also cannot check these links without the content mounted. For example, `/docs/grafana/latest/ from /docs/tempo/latest` won't resolve unless you have both projects mounted in the webserver.
```

**Comment by chri2547** [comment]
Removing myself as you've got other very capable reviewers. Thanks!

File: docs/sources/writing-guide/references/index.md [hunk 9]
```diff
@@ -115,7 +114,29 @@ Refer to the following table for the correct relref to use to link between each
...
+## Versions and cross-repository linking
+
+For Grafana's webserver environments, you can't address other versions of the docs, such as a version-specific archived docs set (`https://grafana.com/docs/grafana/v8.5/` and so forth) or `/next/` docs for links in content residing in `/latest/`, using Hugo references.
+
+Hugo references addressed across different products' docs, such as from `/docs/grafana/` to `/docs/loki/` and vice-versa, as well as references from docs addressed to other Hugo-published content on grafana.com, can also be predictably addressed.
+
+To avoid broken links in these situations on grafana.com, use regular Markdown link syntax (`[link text](/docs/repo/version/folder/file/)`) instead of Hugo references. To ensure the links work in local builds, staging environments, and the live website, you **shouldn't** use a fully qualified URL with `https://grafana.com` for links to other content on grafana.com.
+
+For cross-repository links, use a standard markdown link, with the link structured like so:
+`\docs\repo\page\`
```
Comment (knylander-grafana) [inline docs/sources/writing-guide/references/index.md:126]:
```suggestion
For cross-repository links, use a standard markdown link, with the link structured like this: `\docs\repo\page\`.
```

File: docs/sources/writing-guide/references/index.md [hunk 10]
```diff
@@ -115,7 +114,29 @@ Refer to the following table for the correct relref to use to link between each
...
+For example:
+
+```markdown
+This is an [example cross-repository link](/docs/grafana/whatsnew) to the Grafana repository.
+```
+
+Using a Hugo `relref` in a cross-repository link or a link to a specific version can result in a page not found error message when running `make docs` if the linked content isn't mounted when using the script.
+
+Unlike references, Hugo does _not_ confirm that these link destinations exist during its build, so manually confirm that the published links in a local build and on the published website point correctly.
+With partial URIs, you also cannot check these links without the content mounted. For example, `/docs/grafana/latest/ from /docs/tempo/latest` won't resolve unless you have both projects mounted in the webserver.
```
Comment (jdbaldry) [inline docs/sources/writing-guide/references/index.md:137]:
```suggestion
With partial URIs, you also cannot check these links without the content mounted. For example, `/docs/grafana/latest/` from `/docs/tempo/latest` won't resolve unless you have both projects mounted in the webserver.
```

**Review by jdbaldry — APPROVED** [review]

**Pull request merged**